test/e2e/operator_test.go

```diff
@@ -4247,6 +4251,42 @@ func waitForIngressControllerCondition(t *testing.T, cl client.Client, timeout t
 	return err
 }
 
+// isServiceDeleted checks if the service was deleted and return true if it is not found.
+func isServiceDeleted(t *testing.T, cl client.Client, ic *operatorv1.IngressController, timeout time.Duration) error {
+	lbService := &corev1.Service{}
+	serviceName := controller.LoadBalancerServiceName(ic)
+
+	err := wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		if err := cl.Get(context.TODO(), serviceName, lbService); err != nil {
+			if apierrors.IsNotFound(err) {
+				return true, nil
+			}
+			t.Logf("failed to delete the load balancer service %s/%s for IC %s due to error: %v", serviceName.Namespace, serviceName.Name, ic.Name, err)
+			return false, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return fmt.Errorf("timed out waiting for the load balancer service to be deleted: %v", err)
+	}
+	return nil
+}
+
+// assertEIPAllocationDeleted cleans the EIPs having a tag key and value and the polling to clean EIPs continues until all the unassociated EIPs are released.
+func assertEIPAllocationDeleted(t *testing.T, svc *ec2.EC2, ic *operatorv1.IngressController, timeout time.Duration) {
+	t.Helper()
+	if err := isServiceDeleted(t, kclient, ic, timeout); err == nil {
```

Comment:
see comment regarding this being a part of the IngressController clean up, not the EIP clean up.